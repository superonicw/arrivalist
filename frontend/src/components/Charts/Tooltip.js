import React from 'react'
import { makeStyles } from '@material-ui/core/styles'
import moment from 'moment'
import numeral from 'numeral'

const useStyles = makeStyles(() => ({
  tooltip: {
    color: 'white',
    background: 'rgba(0, 0, 0, 0.9)',
    border: '1px solid transparent',
    borderRadius: 3,
    padding: '4px 8px',
    marginTop: -30,
  },
}))

const Tooltip = ({ x, y, state }) => {
  const classes = useStyles()

  return (
    <div className={classes.tooltip}>
      <div>
        <b>{state}</b>
      </div>
      <div>
        {moment(x).format('MMM Do')}: <b>{numeral(y).format('0,0')}</b>
      </div>
    </div>
  )
}

export default Tooltip
