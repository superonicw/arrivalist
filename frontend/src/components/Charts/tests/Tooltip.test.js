import React from 'react'
import { mount } from 'enzyme'
import Tooltip from '../Tooltip'

const initialProps = {
  x: '2020-09-01',
  y: '12000',
  state: 'AL',
}

describe('Tooltip test', () => {
  it('renders correctly', () => {
    const wrapper = mount(<Tooltip {...initialProps} />)
    expect(wrapper).toMatchSnapshot()
  })
})
