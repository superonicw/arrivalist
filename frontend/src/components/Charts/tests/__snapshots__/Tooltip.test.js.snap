// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip test renders correctly 1`] = `
<Tooltip
  state="AL"
  x="2020-09-01"
  y="12000"
>
  <div
    className="makeStyles-tooltip-1"
  >
    <div>
      <b>
        AL
      </b>
    </div>
    <div>
      Sep 1st
      : 
      <b>
        12,000
      </b>
    </div>
  </div>
</Tooltip>
`;
